src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs

```diff
@@ -231,38 +239,59 @@ public string XmlSchemaCollectionDatabase
         {
             get
             {
-                string xmlSchemaCollectionDatabase = _xmlSchemaCollectionDatabase;
-                return (xmlSchemaCollectionDatabase ?? ADP.StrEmpty);
+                return (_xmlSchemaCollection?.Database ?? ADP.StrEmpty);
             }
             set
             {
-                _xmlSchemaCollectionDatabase = value;
+                bool collectionIsNull = _xmlSchemaCollection != null;
+                if (value != null || collectionIsNull)
+                {
+                    if (collectionIsNull)
+                    {
+                        _xmlSchemaCollection = new SqlMetaDataXmlSchemaCollection();
+                    }
+                    _xmlSchemaCollection.Database = value;
+                }
             }
         }
 
         public string XmlSchemaCollectionOwningSchema
         {
             get
             {
-                string xmlSchemaCollectionOwningSchema = _xmlSchemaCollectionOwningSchema;
-                return (xmlSchemaCollectionOwningSchema ?? ADP.StrEmpty);
+                return (_xmlSchemaCollection?.OwningSchema ?? ADP.StrEmpty);
             }
             set
             {
-                _xmlSchemaCollectionOwningSchema = value;
+                bool collectionIsNull = _xmlSchemaCollection != null;
+                if (value != null || collectionIsNull)
+                {
+                    if (collectionIsNull)
```

Comment:
`if (collectionIsNull)` block can go outside of the check for value!=null as it will executed whatever the value of value

Reply:
Same below

Reply:
I'm not following. The collection is a lazy nullable. 
if value==null && collection ==null pass
If value!=null && collection == null create and set value
if value!=null && collection !=null set value
if value==null && collection != null set value
wouldn't moving inner if statement outside just force create it in a situation where it wasn't needed?

Reply:
@jnm2 talked me through this slowly on gitter. I'm not really a fan of either construction because they both feel clumsy but I can go with the less nested version if it's preferred. Fixed.